scripts/bot-deployment/DeployReporterCR.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+set -e
+
+if [ $# -ne 1 ]; then
+    echo "Incorrect number of arguments supplied! First and only argument is bot's name. From this the bot's config will be inferred."
+    echo "example: ./DeployBotCR.sh ethbtc-mainnet-reporter"
+    echo "To view all available configs run: gsutil ls gs://bot-configs"
+    exit 1
+fi
+
+echo "🔥 Starting GCP Cloud Run deployment script for bot" $1
+
+echo "🤖 Pulling bot config from GCP bucket"
+
+# Create a temp file to store the config. This will be cleaned up by the OS.
+tempFile=$(mktemp -t UMA)
+
+# Copy config files from GCP to the temp file.
+gsutil cp gs://bot-configs/$1.yml $tempFile
+
+echo "✍️  Config has been pulled and placed in a temp directory" $tempFile
+
+echo "🚀 Deploying cloud run instance to GCP"
+
+# Deploy The bot to GCP using the config file.
+gcloud beta run services replace $tempFile \
```

Comment:
Two questions:
1. Does this replace work if there's no existing job?
2. Where does the bot name get fed into this command? It looks like it's referenced in the `services describe` command below? How do those end up matching up?

Reply:
1. Yup. Replace is used if you are deploying a new job or upgrading an existing one with Cloud run. Strange syntax, I know.
2. The name and all other information is fed in from the config file. This has the relevant structure as follows:
```
...
metadata:
  name: ethbtc-mainnet-reporter
...
```
This name here matches to the name of the bot in CLoud run.

You do raise a good point, though. It is possible to name the bot differently to the config file. I'll look into a better way of either validating these match, injecting the name into the config or specifying the name of the bot as a param.

Reply:
I looked a bit further into this and it's not super trivial to use this command in the way I had hoped to validate that the name provided matches the config name. Because we use yaml config files to deploy the cloud run instances (used to store all env variables and cloud run configs) you cant override the `SERVICE` param from a CLI flag.

I've been using this config from the docs [here](https://cloud.google.com/run/docs/deploying#service). From the API [docs](https://cloud.google.com/sdk/gcloud/reference/beta/run/services/replace) we are limited to using only config files to define the `SERVICE` name.

However, I don't really think this is a big deal. As long as the `name` field within the config matches to the actual name of the config file there is no way the names can not line up because the script pulls down the config based on the file name. 

In fact, we have the same restriction with other bot deployment scripts with the `BOT_IDENTIFIER`. There is no validation to check that the name of the bot equals the `BOT_IDENTIFIER`. 

In future we could consider creating a script to validate these configs but for now this would be totally over-engineered imho.

TLDR: we need to always ensure the config names match to the name of the bot in the config itself.